### PR TITLE
Add `resetNashSolver()` to the public API

### LINKED ISSUE #2: Link error for identifier `FirstTime`
In CI [here](https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=10325&view=logs&j=01c8d896-7418-5fed-6445-8dfc3b1e9f99&t=ff333c5e-8323-5ed2-ced7-5e686ce0c0de&l=231), clang complained of a double definition.  

This looks like a valid complaint to me, though as we saw [here](https://github.com/JuliaPackaging/Yggdrasil/pull/2862#issuecomment-822897359), gcc successfully compiles the same code.  Presumably this means the code is incorrect (i.e. compiler behavior is unspecified) but gcc is being lenient in some way.

Quoting @giordano:

> maybe the issue is that the global variable in the header file is being defined twice in two different object files. It's just a completely bad idea to define variables in header files (without being extern at least).

I've personally never seen an issue like this before, for (I think) two reasons: (1) I've never seen a non-`static`, non-`extern` global variable in a header file; (2) all the codebases I've worked in have had include guards on the header files, so re-declaration was not happening.

## What we need to do

We should fix the link error in a way that does not change program behavior.

As far as I can see, the only sane way this program would compile is if the offending [line](https://github.com/JuliaPolyhedra/lrslib/blob/d8b723a2c315614979a8354f9e768d273d14a215/lrsnashlib.h#L70) is being treated as an `extern` declaration.  However I don't have iron-clad proof of this, and I think it's a statement about the behavior of GCC that is not mandated by the C standard.

Of course, we could also start reasoning about the program itself and what it's supposed to do, and/or contact the author :slightly_smiling_face:

## Attempting to get a full explanation

According to [Wikipedia](https://en.wikipedia.org/w/index.php?title=External_variable&oldid=986074372#Definition,_declaration_and_the_extern_keyword):

> If neither the extern keyword nor an initialization value are present, the statement can be either a declaration or a definition. It is up to the compiler to analyse the modules of the program and decide.

According to [cppreference](https://en.cppreference.com/w/c/language/declarations)

> For objects, a declaration that allocates storage (automatic or static, but not extern) is a definition, while a declaration that does not allocate storage (external declaration) is not.
>
> ```c
> extern int n; // declaration
> int n = 10; // definition
> ```

Sadly, the above quote does not explicitly say whether the statement
```c
int n;
```
with no storage-class specifier and no initialization must be considered a definition, or must be considered only a declaration, or whether it's up to the toolchain.  (Wikipedia apparently says it's up to the toolchain.)

As far as I can tell, [this](https://github.com/JuliaPolyhedra/lrslib/blob/d8b723a2c315614979a8354f9e768d273d14a215/lrsnashlib.h#L70) line is getting inlined twice by the preprocessor (since there is no include guard), and the clang error above seems to indicate that clang is considering at least two of the copies of that line to be definitions (since the [only other](https://github.com/JuliaPolyhedra/lrslib/blob/d8b723a2c315614979a8354f9e768d273d14a215/lrsnashlib.c#L51) use of that name in the codebase is definitely neither a definition nor a declaration).  Perhaps gcc chooses not to consider more than one of those copies a definition.

On the other hand, if all copies of that [line](https://github.com/JuliaPolyhedra/lrslib/blob/d8b723a2c315614979a8354f9e768d273d14a215/lrsnashlib.h#L70) are considered to be declarations and not definitions, then the following excerpt from [cppreference](https://en.cppreference.com/w/c/language/storage_duration):

> If no storage-class specifier is provided, the defaults are:
>
> * `extern` for all functions 
> * `extern` for objects at file scope 
> * `auto` for objects at block scope 

implies that the line is equivalent to
```c
extern long FirstTime;
```

### PR DESCRIPTION
**Pull request by bzinberg**
Fixes https://github.com/JuliaPolyhedra/lrslib/issues/2.

## Tested
Builds successfully on my machine.

### CODE REVIEW

File: lrsnashlib.h [hunk 1]
```diff
@@ -67,7 +67,7 @@ void initFwidth(game *g);
 void updateFwidth(game *g, int col, int pos, char *str);
 
 
-long FirstTime;          /* set this to true for every new game to be solved */
+extern long FirstTime;   /* set this to true for every new game to be solved */
```
Comment (giordano) [inline lrsnashlib.h]: Why defining it here at all?

Comment (bzinberg) [reply lrsnashlib.h]: My reading of the standard is that every variable must be defined exactly once, and an `extern` declaration is not a definition.

Comment (giordano) [reply lrsnashlib.h]: So you can remove this line :slightly_smiling_face:

Comment (bzinberg) [reply lrsnashlib.h]: then the variable would have internal linkage

**Comment by blegat** [comment]
Can you redirect the change to the branch of https://github.com/JuliaPolyhedra/lrslib/pull/4 ? We won't be making change for the old version of lrslib, let's focus on v7.1a. There the definition in the `.h` was removed and we need to add it back, see https://github.com/JuliaPolyhedra/LRSLib.jl/pull/41#discussion_r618159018

**Comment by blegat** [comment]
Probably even better is to add a function for modifying `FirstTime` and not defining the variable in `.h`.

**Comment by bzinberg** [comment]
> Probably even better is to add a function for modifying `FirstTime` and not defining the variable in `.h`.

Responded to this in https://github.com/JuliaPolyhedra/LRSLib.jl/pull/41#discussion_r618515852

**Comment by bzinberg** [comment]
> Can you redirect the change to the branch of #4 ? We won't be making change for the old version of lrslib, let's focus on v7.1a. There the definition in the `.h` was removed and we need to add it back, see [JuliaPolyhedra/LRSLib.jl#41 (comment)](https://github.com/JuliaPolyhedra/LRSLib.jl/pull/41#discussion_r618159018)

Absolutely.  Thanks for working on the migration!

**Comment by bzinberg** [comment]
> Can you redirect the change to the branch of #4 ?

Done, and changed the target branch of this PR accordingly.

**Review by blegat — APPROVED** [review]
Thanks!

**Pull request merged**